### PR TITLE
feat(LicenseDataModel): Merge LiceneObligation with Obligation

### PR DESCRIPTION
**Pull request by smrutis1**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.

 * Which issue is this pull request belonging to and how is it solving it? (*#989*)
 * Did you add or update any new dependencies that are required for your change? -No

Issue: Fixes: #989 

### How To Test?

* Need to test the license import and export functionality and adding obligation to licenses

### Checklist
Must:
- [x] All related issues are referenced in commit messages and in PR

Signed-off-by: Smruti Prakash Sahoo <smruti.sahoo@siemens.com>

### CODE REVIEW

File: scripts/migrations/030_obligation_field_update.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+# ---------------------------------------
+
+def removeFieldName(qryResult, fieldToBeRemoved, log):
+    print 'Removing field name '+fieldToBeRemoved+' starts'
+    log['Updated obligation fields '+fieldToBeRemoved] = []
+    for entity in qryResult:
+        del entity[''+fieldToBeRemoved+'']
+        if not DRY_RUN:
+            db.save(entity)
+            print 'Removing field name '+fieldToBeRemoved+' done'
```
Comment (JaideepPalit) [inline scripts/migrations/030_obligation_field_update.py:59]: Maybe more info can be added to this statement like Id.
Current O/P
```
Getting all obligation with field obligationDatabaseIds
found 7 obligation with field obligationDatabaseIds in db!
Removing field name obligationDatabaseIds starts
Removing field name obligationDatabaseIds done
Removing field name obligationDatabaseIds done
Removing field name obligationDatabaseIds done
Removing field name obligationDatabaseIds done
Removing field name obligationDatabaseIds done
Removing field name obligationDatabaseIds done
Removing field name obligationDatabaseIds done
```

**Review by JaideepPalit — CHANGES_REQUESTED** [review]
Kindly look into some minor review comments.
Tested the PR seems to be some issue while adding License obligation from License

image

File: backend/src/src-licenses/src/test/java/org/eclipse/sw360/licenses/TestLicenseClient.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
 import com.google.common.collect.ImmutableList;
 import org.eclipse.sw360.datahandler.thrift.licenses.License;
 import org.eclipse.sw360.datahandler.thrift.licenses.LicenseService;
-import org.eclipse.sw360.datahandler.thrift.licenses.LicenseObligation;
+/*import org.eclipse.sw360.datahandler.thrift.licenses.LicenseObligation;*/
```
[the remark this answers is not in the document]
Comment (smrutis1) [reply backend/src/src-licenses/src/test/java/org/eclipse/sw360/licenses/TestLicenseClient.java]: done

File: frontend/sw360-portlet/src/main/resources/content/Language.properties [hunk 3]
```diff
@@ -1138,7 +1138,7 @@ to.be.replaced=To be replaced
 obligation=Obligation
 obligation.details=Obligation Details
 obligations=Obligations
-obligations.and.obligations=Todos and Obligations
+#obligations.and.obligations=Todos and Obligations
```
[the remark this answers is not in the document]
Comment (smrutis1) [reply frontend/sw360-portlet/src/main/resources/content/Language.properties]: done

File: libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/thrift/ThriftUtils.java [hunk 4]
```diff
@@ -55,7 +55,8 @@
             .add(AttachmentContent.class) // Attachment service
             .add(AttachmentUsage.class) // Attachment service
             .add(Component.class).add(Release.class) // Component service
-            .add(License.class).add(Obligation.class).add(LicenseObligation.class) // License service
+            .add(License.class).add(Obligation.class)
+//            .add(LicenseObligation.class) // License service
```
[the remark this answers is not in the document]
Comment (smrutis1) [reply libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/thrift/ThriftUtils.java]: done

File: scripts/migrations/030_obligation_field_update.py (hunk 1 above)
Comment (smrutis1) [reply scripts/migrations/030_obligation_field_update.py]: appended the `id`

**Comment by smrutis1** [comment]
Thanks for review. Incorporated the review comments along with that addressed the issue related to blank obligation title after adding obligation to title and added a new column(`Obligation Text`) to make it look more verbose and for easy differentiation.

**Comment by mcjaeger** [comment]
(tested on stage)

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/licenses/LicensesPortlet.java [hunk 5]
```diff
@@ -405,46 +410,55 @@ public void addObligations(ActionRequest request, ActionResponse response) throw
...
+                        obl.setDevelopment(false);
+                        obl.setDistribution(false);
+                    }
+                    RequestStatus requestStatus = client.addObligationsToLicense(obl, licenseID, user);
+                    setSessionMessage(request, requestStatus, "License", "update");
+                }
+            } else {
+                List<Obligation> obls = client.getObligationsByIds(Arrays.asList(obligationIds));
+                obls.stream().filter(Objects::nonNull).forEach(obl -> wrapException(() -> {
+                    Obligation oblCopy = obl.deepCopy();
```
Comment (mcjaeger) [inline frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/licenses/LicensesPortlet.java:446]: that seems to be the only relevant part for the if-else

Comment (smrutis1) [reply frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/licenses/LicensesPortlet.java]: refactored the code.

**Review by JaideepPalit — APPROVED** [review]
Code looks good to me

**Pull request merged**